readthedocs/search/views.py

```diff
@@ -32,23 +32,64 @@
         'version',
         'language',
         'role_name',
-        'index',
     ),
 )
 
 
-class SearchView(View):
+class SearchViewBase(View):
 
-    """
-    Global user search on the dashboard.
+    http_method_names = ['get']
+    max_search_results = 50
+
+    def _search(self, user_input, use_advanced_query):
+        """Return search results and facets given a `user_input` to filter by."""
+        if not user_input.query:
+            return [], {}
+
+        filters = {}
+        for avail_facet in ALL_FACETS:
+            value = getattr(user_input, avail_facet, None)
+            if value:
+                filters[avail_facet] = value
+
+        search_facets = {
+            'project': ProjectSearch,
+            'file': PageSearch,
+        }
+        faceted_search_class = search_facets.get(
+            user_input.type,
+            ProjectSearch,
+        )
+        search = faceted_search_class(
+            query=user_input.query,
+            filters=filters,
+            user=self.request.user,
+            use_advanced_query=use_advanced_query,
+        )
+        results = search[:self.max_search_results].execute()
+        facets = results.facets
+
+        # Make sure the selected facets are displayed,
+        # even when they return 0 results.
+        for facet in facets:
+            value = getattr(user_input, facet, None)
+            if value and value not in (name for name, *_ in facets[facet]):
+                facets[facet].insert(0, (value, 0, True))
 
-    This is for both the main search and project search.
+        return results, facets
+
+
+class ProjectSearchView(SearchViewBase):
 
-    :param project_slug: Sent when the view is a project search
     """
+    Search view of the ``search`` tab.
 
-    http_method_names = ['get']
-    max_search_results = 50
+    Query params:
+
+    - q: search term
+    - version: version to filter by
+    - role_name: sphinx role to filter by
```

Comment:
In the `search` tab we only search for files under that project, so no need to choose between serializes to pass any other additional inputs